hack/lib/golang.sh

```diff
@@ -283,32 +269,53 @@ EOF
   if [[ "${TRAVIS:-}" != "true" ]]; then
     local go_version
     go_version=($(go version))
-    if [[ "${go_version[2]}" < "go1.4" ]]; then
+    if [[ "${go_version[2]}" < "go1.6" ]]; then
       kube::log::usage_from_stdin <<EOF
 Detected go version: ${go_version[*]}.
-Kubernetes requires go version 1.4 or greater.
-Please install Go version 1.4 or later.
+Kubernetes requires go version 1.6 or greater.
+Please install Go version 1.6 or later.
 EOF
-      exit 2
+      return 2
     fi
   fi
+}
+
+# kube::golang::setup_env will check that the `go` commands is available in
+# ${PATH}. If not running on Travis, it will also check that the Go version is
+# good enough for the Kubernetes build.
+#
+# Inputs:
+#   KUBE_EXTRA_GOPATH - If set, this is included in created GOPATH
+#
+# Outputs:
+#   env-var GOPATH points to our local output dir
+#   env-var GOBIN is unset (we want binaries in a predictable place)
+#   env-var GO15VENDOREXPERIMENT=1
+#   current directory is within GOPATH
+kube::golang::setup_env() {
+  kube::golang::verify_go_version
+
+  kube::golang::create_gopath_tree
 
   GOPATH=${KUBE_GOPATH}
 
   # Append KUBE_EXTRA_GOPATH to the GOPATH if it is defined.
   if [[ -n ${KUBE_EXTRA_GOPATH:-} ]]; then
     GOPATH="${GOPATH}:${KUBE_EXTRA_GOPATH}"
   fi
-
-  # Append the tree maintained by `godep` to the GOPATH unless KUBE_NO_GODEPS
-  # is defined.
-  if [[ -z ${KUBE_NO_GODEPS:-} ]]; then
-    GOPATH="${GOPATH}:${KUBE_ROOT}/Godeps/_workspace"
-  fi
   export GOPATH
 
+  # Change directories so that we are within the GOPATH.  Some tools get really
+  # upset if this is not true.
+  local subdir
+  subdir=$(kube::realpath . | sed "s|$KUBE_ROOT||")
+  cd "${KUBE_GOPATH}/src/${KUBE_GO_PACKAGE}/${subdir}"
```

Comment:
There should be no .. in the path at all.  First it does cd $(realpath .),
then changes to ./_output/local/go/src/...

On Thu, Apr 28, 2016 at 3:16 PM, Eric Paris <notifications@github.com>
wrote:

> In hack/lib/golang.sh
> <https://github.com/kubernetes/kubernetes/pull/24242#discussion_r61511614>
> :
>
> >    export GOPATH
> >
> > +  # Change directories so that we are within the GOPATH.  Some tools get really
> > +  # upset if this is not true.
> > +  local subdir
> > +  subdir=$(kube::realpath . | sed "s|$KUBE_ROOT||")
> > +  cd "${KUBE_GOPATH}/src/${KUBE_GO_PACKAGE}/${subdir}"
>
> Are there and ../ in the path? does it work if you use cd $(realpath
> $PATHNAME)
>
> —
> You are receiving this because you were mentioned.
> Reply to this email directly or view it on GitHub
> <https://github.com/kubernetes/kubernetes/pull/24242/files/71a332d7a1bad69f61d141a83280cdf5b804ee77#r61511614>
>